src/components/reusable/table/table-head.ts

```diff
@@ -15,6 +15,23 @@ import styles from './table-head.scss';
 export class TableHead extends LitElement {
   static override styles = [styles];
 
+  override firstUpdated() {
+    this.addEventListener('on-sort-changed', this.handleChildSort as EventListener);
+  }
+
+  private handleChildSort(e: CustomEvent) {
+    const sortedColumnKey = e.detail.sortKey;
+
+    // Get all kyn-th elements
+    const allHeaders = Array.from(this.querySelectorAll('kyn-th'));
```

Comment:
I'm surprised this works, since the `kyn-th` are slotted. Normally you would want to use this Lit decorator: https://lit.dev/docs/api/decorators/#queryAssignedElements